app/public/model/data/FetchedDataManager.js

```diff
@@ -135,7 +134,8 @@ export default class FetchedDataManager {
 
     changeRowsOnSite(rowsOnSite) {
         const url = this.router.getUrl();
-        const newUrl = replaceUrlParams(url, [[dataReqParams.rowsOnSite, rowsOnSite]]);
+        this.rowsOnSite = rowsOnSite;
+        const newUrl = replaceUrlParams(url, [[dataReqParams.rowsOnSite, this.rowsOnSite]]);
```

Comment:
[[dataReqParams.rowsOnSite, this.rowsOnSite]] better to replace as dict

Reply:
Refactored in a way that now the object is being passed. I believe that is better.